.gitlab/package_build/linux.yml

```
@@ -134,3 +134,43 @@ iot-agent-armhf:
     # if we use too many compression threads or a too agressive level
     FORCED_PACKAGE_COMPRESSION_LEVEL: 5
 
+.dogstatsd_build_common:
+  rules:
+    - !reference [.except_mergequeue]
+    - when: on_success
+  stage: package_build
+  script:
+    - source /root/.bashrc
+    - echo "About to build for $RELEASE_VERSION"
+    - !reference [.setup_ruby_mirror_linux]
+    - !reference [.setup_python_mirror_linux]
+    - !reference [.retrieve_linux_go_deps]
+    - !reference [.cache_omnibus_ruby_deps, setup]
+    # remove artifacts from previous pipelines that may come from the cache
+    - rm -rf $OMNIBUS_PACKAGE_DIR/*
```

Isn't it better like this ? I think there is a bash error if the directory is empty because of empty glob
```suggestion
    - rm -rf $OMNIBUS_PACKAGE_DIR
```

---

I'd tend to agree with you, however I'm not sure how omnibus will behave if the package folder doesn't exist.
This pattern has been used for as long as I can remember, so I think it's safe (for our purposes at least), but if I'm being honest, I don't understand why it doesn't fail as that folder is expected to be empty AFAICS

---

There's no bash error for that (at least I couldn't reproduce that). I'm not even sure the folder even ever exists at this point, the comment mentions a cache and previous pipelines but I'm not sure that applies to the current code.